com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariableBase.cs

```diff
@@ -71,12 +120,16 @@ public interface INetworkVariable
         /// <param name="keepDirtyDelta">Whether or not the delta should be kept as dirty or consumed</param>
         /// <param name="localTick">The local network tick at which this var was written, on the machine it was written </param>
         /// <param name="remoteTick">The remote network tick at which this var was sent by the host </param>
-        void ReadDelta(Stream stream, bool keepDirtyDelta);
+        public abstract void ReadDelta(Stream stream, bool keepDirtyDelta);
 
         /// <summary>
         /// Sets NetworkBehaviour the container belongs to.
         /// </summary>
         /// <param name="behaviour">The behaviour the container behaves to</param>
-        void SetNetworkBehaviour(NetworkBehaviour behaviour);
+        public void SetNetworkBehaviour(NetworkBehaviour behaviour)
```

Comment:
I think this should be an `internal` because we will be setting-up/registering NetVars on MBs and we wouldn't want users to mess up with it then potentially break things at runtime.

Usage in `NetworkBehaviour.InitializeVariables()`:
```cs
internal void InitializeVariables()
{
    if (m_VarInit)
    {
        return;
    }

    m_VarInit = true;

    FieldInfo[] sortedFields = GetFieldInfoForType(GetType());

    for (int i = 0; i < sortedFields.Length; i++)
    {
        Type fieldType = sortedFields[i].FieldType;

        if (fieldType.IsSubclassOf(typeof(NetworkVariableBase)))
        {
            var instance = (NetworkVariableBase)sortedFields[i].GetValue(this);

            if (instance == null)
            {
                instance = (NetworkVariableBase)Activator.CreateInstance(fieldType, true);
                sortedFields[i].SetValue(this, instance);
            }

            /* ---- WE USE IT HERE ---- */
            instance.SetNetworkBehaviour(this);

            var instanceNameProperty = fieldType.GetProperty(nameof(NetworkVariableBase.Name));
            var sanitizedVariableName = sortedFields[i].Name.Replace("<", string.Empty).Replace(">k__BackingField", string.Empty);
            instanceNameProperty?.SetValue(instance, sanitizedVariableName);

            NetworkVariableFields.Add(instance);
        }
    }

    // ...
```

Reply:
or maybe convert `m_NetworkBehaviour` to be `internal NetworkBehaviour NetworkBehaviour { get; internal set; }` then remove this method entirely :)

Reply:
actually, let me change things in one single commit and you could take it back if you'd like @mattwalsh-unity

Reply:
Yes, I like the internal-ization and yes I like the gettersetter